drivers/usb/device/usb_dc_stm32.c

```diff
@@ -263,6 +264,45 @@ static int usb_dc_stm32_clock_enable(void)
 	return 0;
 }
 
+#if defined(USB_OTG_FS) || defined(USB_OTG_HS)
+static u32_t usb_dc_stm32_get_maximum_speed(void)
+{
+	u32_t speed = USB_OTG_SPEED_FULL;
+
+#ifdef CONFIG_USB_MAXIMUM_SPEED
+
+	/* fail to compile if unsupported speed is defined in device tree */
+#if (CONFIG_USB_MAXIMUM_SPEED != USB_DEVICE_MAX_SPEED_HIGH) && \
+	(CONFIG_USB_MAXIMUM_SPEED != USB_DEVICE_MAX_SPEED_FULL) && \
+	(CONFIG_USB_MAXIMUM_SPEED != USB_DEVICE_MAX_SPEED_LOW)
+#error "Unsupported maximum speed defined in device tree"
+#endif
+
+	if (CONFIG_USB_MAXIMUM_SPEED == USB_DEVICE_MAX_SPEED_HIGH) {
+		speed = USB_OTG_SPEED_HIGH;
+	} else if (CONFIG_USB_MAXIMUM_SPEED == USB_DEVICE_MAX_SPEED_FULL) {
+#if defined(USB_HS_PHYC) && defined(CONFIG_USB_HS_BASE_ADDRESS)
+		speed = USB_OTG_SPEED_HIGH_IN_FULL;
+#else
+		speed = USB_OTG_SPEED_FULL;
+#endif /* USB_HS_PHYC && CONFIG_USB_HS_BASE_ADDRESS */
+	} else if (CONFIG_USB_MAXIMUM_SPEED == USB_DEVICE_MAX_SPEED_LOW) {
+		speed = USB_OTG_SPEED_LOW;
+	}
+#else
+	/*
+	 * If max-speed is not passed via DT, set it to USB controller's
+	 * maximum hardware capability.
+	 */
+#if defined(USB_HS_PHYC) && defined(CONFIG_USB_HS_BASE_ADDRESS)
```

Comment:
Small nitpick, it might be a good idea to move that code at the beginning of the function when `USB_OTG_SPEED_FULL` is assigned to `speed`.

Reply:
Done